master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -674,15 +674,11 @@ private[celeborn] class Master(
         else Math.min(slotsAssignMaxWorkers, requestSlots.maxWorkers)),
       numAvailableWorkers)
     val startIndex = Random.nextInt(numAvailableWorkers)
```

Comment:
Looks clearer now. BTW, maybe here we need a zero check for `numAvailableWorkers`.

Reply:
Thanks, updated

Reply:
Although it is now easy to read, this approach has memory inefficient drawback.

Reply:
@RexXiong @FMX 
thanks for share your worry, had update the patch with using `for` directly, the complexity of the algorithm is o (1).

Reply:
+1, LGTM